src/neo/Plugins/Plugin.cs

```diff
@@ -143,18 +123,12 @@ protected IConfigurationSection GetConfiguration()
             return new ConfigurationBuilder().AddJsonFile(ConfigFile, optional: true).Build().GetSection("PluginConfiguration");
         }
 
-        protected static T GetService<T>()
-        {
-            return services.OfType<T>().FirstOrDefault();
-        }
-
         private static void LoadPlugin(Assembly assembly)
         {
             foreach (Type type in assembly.ExportedTypes)
             {
                 if (!type.IsSubclassOf(typeof(Plugin))) continue;
                 if (type.IsAbstract) continue;
-                if (!CheckRequiredServices(type)) continue;
 
                 ConstructorInfo constructor = type.GetConstructor(Type.EmptyTypes);
```

Comment:
Constructor with `NeoSystem`?

Reply:
You can't assume the constructor of all plugins type.

Reply:
We can detect if it has a `NeoSystem` and send it.

Reply:
But we need to add the constructor to all the plugins. This way we only set the property after initialization.

Reply:
I got it wrong. we can not sent `NeoSystem` to the constructor because all systems share the same plugins.